internal/pkg/manifest/workload_test.go

```diff
@@ -649,7 +651,117 @@ func TestNetworkConfig_IsEmpty(t *testing.T) {
 	}
 }
 
+func TestSecurityGroupsConfig_UtilityFuncForSecurityGroups(t *testing.T) {
+	testCases := map[string]struct {
+		data                 SecurityGroupsIDsOrConfig
+		wantedSecurityGroups []string
+	}{
+		"nil returned when no security groups are specified": {
+			data:                 SecurityGroupsIDsOrConfig{},
+			wantedSecurityGroups: nil,
+		},
+		"security groups in map are returned": {
+			data: SecurityGroupsIDsOrConfig{
+				AdvancedConfig: SecurityGroupsConfig{
+					SecurityGroups: []string{"group", "group1"},
+				},
+			},
+			wantedSecurityGroups: []string{"group", "group1"},
+		},
+		"nil returned when security groups in map are empty": {
+			data: SecurityGroupsIDsOrConfig{
+				AdvancedConfig: SecurityGroupsConfig{
+					SecurityGroups: []string{},
+				},
```

Comment:
also just nit: would this be a better example of what could take place?
```suggestion
				AdvancedConfig: SecurityGroupsConfig{
					DenyDefault: aws.Bool(bool)
				},
```

Since in go, a field of type `[]sting` gets initialized as `nil` instead of `[]string{}`, a non-nil slice with a length of 0.